pkg/test/framework/components/stackdriver/kube.go

```diff
@@ -133,6 +135,42 @@ func (c *kubeComponent) ListTimeSeries() ([]*monitoringpb.TimeSeries, error) {
 	return ret, nil
 }
 
+func (c *kubeComponent) ListLogEntries() ([]*loggingpb.LogEntry, error) {
+	client := http.Client{
+		Timeout: 5 * time.Second,
+	}
+	resp, err := client.Get("http://" + c.forwarder.Address() + "/logentries")
+	if err != nil {
+		return []*loggingpb.LogEntry{}, err
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return []*loggingpb.LogEntry{}, err
+	}
+	var r loggingpb.ListLogEntriesResponse
+	err = jsonpb.UnmarshalString(string(body), &r)
+	if err != nil {
+		return []*loggingpb.LogEntry{}, err
+	}
+	var ret []*loggingpb.LogEntry
+	for _, l := range r.Entries {
+		// Remove fields that do not need verification
+		l.Timestamp = nil
+		l.Severity = ltype.LogSeverity_DEFAULT
+		l.HttpRequest.ResponseSize = 0
+		l.HttpRequest.RequestSize = 0
+		l.HttpRequest.ServerIp = ""
+		l.HttpRequest.RemoteIp = ""
+		l.HttpRequest.Latency = nil
+		delete(l.Labels, "request_id")
+		delete(l.Labels, "source_name")
```

Comment:
just curious, what are the values in this?

Reply:
They are pod names. We could verify them, it requires some change to the echo app interface though.